README
Lingua-YaTeA version 0.4
================================

(Yet Another Term ExtrActor)

YaTeA is a module implementing a configurable term extractor. It
detecting term candidates in corpora. It can also take into account
testified terms issued from terminological resources.

Contains this script :

    yatea

(see MANIFEST for all the files)

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

Or if you don't have the "make" program:

   perl Build.PL
   Build
   Build test
   Build install

 It requires Mdoule::Build version 0.28 (at least) and File::chmod


DEPENDENCIES

This module requires these other modules and libraries:

 Pod::Usage
 Getopt::Long
 Config::General >= 2.30
 Data::Dumper
 File::Path
 Parse::Yapp
 File::Compare
 Pod::Usage
 Test::More


DOCUMENTATION

See the documentation in the module (man Lingua::YaTeA)

COPYRIGHT AND LICENCE

Copyright (C) 2005 by Thierry Hamon and Sophie Aubin

This library and this program are free software; you can redistribute
them and/or modify it under the same terms as Perl itself, either Perl
version 5.8.6 or, at your option, any later version of Perl 5 you may
have available.




Changes
Revision history for Perl extension Lingua::YaTeA.

0.621   - Add the installation of the FR-Flemm files (which have been
          forgot before:-(
        - Add pod directive "=encoding utf8" to manage utf8 encoding


0.62    - better management of argument in the script yatea
	- Correction in utf8 use directive to avoid failure with CPAN
          tests
	- correction deprecated code in IndexSet (defined @array)

0.61    - Add use utf8 in the script yatea to avoid failure with CPAN
          tests
        - Minor correction in the documentation of the main module

0.6     
        - Add of installation tests 
        - Add of examples for French
        - Integration of the MIG corrections (Node, Testified terms)
        - Correction in the forbidden structure management (split
          action)
        - Add NEXT.pm in the pre-required module list
        - UTF-8 is used as charset for both configuration files and
          texts to process
        - Integration of attested terms (in a bracketed format, can be
          produced by the bootstrap option)
        - Option "bootstrap" to generate an output in a bracketed
          format. This output can be used as a attested resource on a
          other corpus
        - Add a defined test in the function appendInclude (but it
          seems there is still a bug on tree building).
        - Monolexical term occurrences are identified as MNP when the
          monolexical oprion is set
        - Addition of a config set for French based on the POSTagger
          Flemm (which uses the Multex tagset) and modification of the
          read method in Corpus.pm (addition opf the parameter
          language) to normalisze the input (correction of the output
          of Flemm)
	- Correction in the input normalization for Flemm
        - Each printing function can now print results on stdout 
        - Replace ':' as tag by 'COLUMN'
        - Add of some Chunking Frontiers as '('
	- Only the tag is taken into account for sentence_boundary
          detection
	- Add Weights section to the DTD and in the XML rendering
	- Correction: TF-IDF based ranking method is a DDW ranking
	- Add ROOT as a reference to the term in which the current
          term is nested
        - Addition of several term weighting and selection measures
          (C-Value and variations, iLong, ilnc, iLong, and term
          autonomy)
        - Add the status of the term candidate (0 : not a term, 1 :
          term). By default, term candidate are terms
        - Add a option for term output style to print only term
          candidate having the status of term
        - colors of the terms in the HTML output can be parametrized
          (option file, options PARSED_COLOR and UNPARSED_COLOR)
	- new function for printing only list of candidate terms
          without XML header
        - Addition of the option XML-corpus-raw, rendering the corpus
          in XML format with terms. Documents and sentences are
          identified in the XML files.


0.5     - Minor correction in the installation process
        - for some OS, Config::General returns while setting variables
          as yatea.rc is on read-inly mode

0.4     - use warnings in all the files
        - correction of minor bugs
        - control of the input type in Node.pm,  NodeSet.pm, Tree.pm
        - bug fix in Tree.pm (method
          completeDiscontinuousNodes). Initialisation of
          a variable
        - carriage returns are deleted in the loaded files (module
          ForbiddenStructureSet.pm, method CleanInfo, and module
          ChunkingDataSet.pm, method LoadData,Corpus.pm:WrapBlocks)
	- addition of the option "TC_MAXIMUM_LENGTH" to allow to limit
          the length of the extracted term candidates (set in
          configuration file Options, default is 12 words)
	- homogeneisation of sub-trees attachement
	- addition of a TF-IDF based ranking method for extracted term
          candidates
        - Workaround for a missing feature in Module::Info : if no
          prefix is set while running "perl Build.PL", additionnal
          directories 'etc' and 'conf' are not installed (or well installed)
        - Modification of Makefile.PL to use the Module::Build (it
          should be a CPAN.pm-compatible distribution)
	- Variables CONFIG_REP, LOCALE_REP, RESULT_REP (yatea.rc) are
          renamed CONFIG_DIR, LOCALE_DIR, RESULT_DIR respectively. 
	- Variables CONFIG_DIR, LOCALE_DIR, RESULT_DIR (yatea.rc) are
          automatically set, while the configuration (in Build.PL)
        - The location of the rc file (yatea.rc) is automatically set,
          while the configuration (in Build.PL)
	- documentation of the module TagSet.pm, File.pm and FileSet.pm
	- Corrections in the LICENSE file

0.3     - removing the .config file
        - bug fix: check in the hook_node is defined in Node.pm
        - replacement of the use of the Parse::Lex module by a
	  module generated from Parse::Yapp
        - addition of examples in the YaTeA PM
        - correction of a bug when calculating the offset of the end
	  char in component of term (especially the left components)
        - in parseProgressively, taking into account that the unparsed
	  array can be empty


0.2     - correction of bugs (or temporary workaround) in the
          tree/node management
        - Removing the file Config.pm (no more necessary)
          and from Makefile.PL
        - YaTeA.pm has been moved in Lingua (before it was in Lingua/YaTeA)
        - Addition of a Build.PL file        
	- addition of POD tests (pod.t, pod-coverage.t)

0.1  
	- Version of YaTeA with a CPAN configuration



Build.PL
use Module::Build;

# my $class = Module::Build->subclass(code => <<'EOF');
# sub process_dat_files {
#     my $self = shift;
#     ... locate and process *.dat files,
#     ... and create something in blib/lib/
#     }
# EOF


 
my $class = Module::Build->subclass(
				    class => "Module::Build::Custom",
code => <<'SUBCLASS' );

sub ACTION_install {
    my $self = shift;
    $self->SUPER::ACTION_install;
#     warn "\nDon't forget to set the variables CONFIG_DIR and LOCALE_DIR in the configuration file yatea.rc to the right path if necessary.\n\n";
}

sub ACTION_fakeinstall {
    my $self = shift;
    $self->SUPER::ACTION_fakeinstall;

#     warn "\nDon't forget to set the variables CONFIG_DIR and LOCALE_DIR in the configuration file yatea.rc to the right path if necessary.\n\n";
}

sub ACTION_build {
    my $self = shift;

    require Config::General;

    warn "Setting configuration variables in etc/yatea/yatea.rc\n";
    my $rcfile = $self->base_dir() . "/etc/yatea/yatea.rc";
    # warn "RCFILE: $rcfile\n";
    chmod(0644, $rcfile);
    if (!( -f "$rcfile.orig") ) {
	File::Copy::copy($rcfile, "$rcfile.orig");
    }

    my $conf = new Config::General('-ConfigFile' => $rcfile . ".orig",
				   '-InterPolateVars' => 0,
				   '-InterPolateEnv' => 0,
				   '-StoreDelimiter' => " = ",
				   );
    my %config = $conf->getall;
    # warn "SHARE: " .  $self->install_destination("share") . "\n";
    $config{'DefaultConfig'}->{'CONFIG_DIR'} = $self->install_destination("share") . "/YaTeA/config";
    $config{'DefaultConfig'}->{'LOCALE_DIR'} = $self->install_destination("share") . "/YaTeA/locale";

    # warn $config{'DefaultConfig'}->{'CONFIG_DIR'} . "\n";

    $conf->save_file($rcfile,\%config);
    warn "Done\n";

    warn "Setting the default location of yatea.rc in lib/Lingua/YaTeA.pm\n";
    my $mainpmfile = $self->base_dir() . "/lib/Lingua/YaTeA.pm";
    
    require File::Copy;

    my $n = FileHandle->new($mainpmfile,"r");
#     File::Copy->cp($n,"$mainpmfile.orig");

    chmod(0644, $mainpmfile);
    if (!( -f "$mainpmfile.orig") ) {
	File::Copy::copy($mainpmfile, "$mainpmfile.orig");
    }
    my $line;
    open MAINPMFILE, "$mainpmfile.orig" or die "No such file ($mainpmfile.orig)\n";
    open MAINPMFILEDEST, ">$mainpmfile" or die "Can not open file $mainpmfile\n";
    while($line = <MAINPMFILE>) {
	if ($line =~ /^(\s\$rcfile\s=\s\")\/etc\/yatea\/yatea.rc(\".*\n)/) {
	    $line = $1 . $self->install_destination("conf") . "/yatea/yatea.rc" . $2;
	}
	print MAINPMFILEDEST $line;
    }
    close MAINPMFILE;
    close MAINPMFILEDEST;
    warn "Done\n";

    $self->SUPER::ACTION_build;
}

sub ACTION_clean {
    my $self = shift;
    $self->SUPER::ACTION_clean;

    require File::Copy;
    my $mainpmfile = $self->base_dir() . "/lib/Lingua/YaTeA.pm";
    File::Copy::move("$mainpmfile.orig", $mainpmfile);

    my $rcfile = $self->base_dir() . "/etc/yatea/yatea.rc";
    File::Copy::move("$rcfile.orig", $rcfile);
}

SUBCLASS

my $build = $class->new
    (
     module_name => 'Lingua::YaTeA',
     dist_version_from => 'lib/Lingua/YaTeA.pm',
     license => 'perl',
     configure_requires => { 'Module::Build' => 0.38,
			     'File::chmod' => 0,
     },
     requires => {'perl' => '5.005',
		  Parse::Yapp => 0,
		  Getopt::Long => 0,
		  File::Compare => 0,
		  Data::Dumper => 0, 
		  Pod::Usage => 0,
		  Config::General => 2.30,
		  Test::More     => 0.62,
		  NEXT => 0,
		  Encode => 0,
  	 	  File::Path => 0,
		  POSIX => 0,
		  Scalar::Util => 0,
		  Test::Pod::Coverage => 0,
		  Test::Pod => 0,
		  },

#      $build->add_build_element('dat');
#      $build->create_build_script;

    build_requires => {
        "Module::Build"     => 0.28,
	Test::More     => 0.62,
    },


#       installdirs => 'core',
     conf_files => {'etc/yatea/yatea.rc' => 'conf/yatea/yatea.rc'},
     share_files => {'share/doc/YaTeA/affichage.fig' => 'share/doc/YaTeA/affichage.fig',
		     'share/doc/YaTeA/alvis.sty' => 'share/doc/YaTeA/alvis.sty',
		     'share/doc/YaTeA/analyseDefinis.fig' => 'share/doc/YaTeA/analyseDefinis.fig',
		     'share/doc/YaTeA/AnalyseProgressive.fig' => 'share/doc/YaTeA/AnalyseProgressive.fig',
		     'share/doc/YaTeA/analyseSSDefinis.fig' => 'share/doc/YaTeA/analyseSSDefinis.fig',
		     'share/doc/YaTeA/ArchitectureDescrYatea.tex' => 'share/doc/YaTeA/ArchitectureDescrYatea.tex',
		     'share/doc/YaTeA/chargementPatrons.fig' => 'share/doc/YaTeA/chargementPatrons.fig',
		     'share/doc/YaTeA/chargementTAtt.fig' => 'share/doc/YaTeA/chargementTAtt.fig',
		     'share/doc/YaTeA/ChercheIlots.fig' => 'share/doc/YaTeA/ChercheIlots.fig',
		     'share/doc/YaTeA/ct.fig' => 'share/doc/YaTeA/ct.fig',
		     'share/doc/YaTeA/DataDescrYatea.tex' => 'share/doc/YaTeA/DataDescrYatea.tex',
		     'share/doc/YaTeA/decoupage.fig' => 'share/doc/YaTeA/decoupage.fig',
		     'share/doc/YaTeA/DTD/yatea.dtd' => 'share/doc/YaTeA/DTD/yatea.dtd',
		     'share/doc/YaTeA/endogenous.fig' => 'share/doc/YaTeA/endogenous.fig',
		     'share/doc/YaTeA/entreeLex.fig' => 'share/doc/YaTeA/entreeLex.fig',
		     'share/doc/YaTeA/exogenous.fig' => 'share/doc/YaTeA/exogenous.fig',
		     'share/doc/YaTeA/foret.fig' => 'share/doc/YaTeA/foret.fig',
		     'share/doc/YaTeA/FuntionDescrYatea.tex' => 'share/doc/YaTeA/FuntionDescrYatea.tex',
		     'share/doc/YaTeA/generalExtracteur.fig' => 'share/doc/YaTeA/generalExtracteur.fig',
		     'share/doc/YaTeA/gnms.fig' => 'share/doc/YaTeA/gnms.fig',
		     'share/doc/YaTeA/ilot.fig' => 'share/doc/YaTeA/ilot.fig',
		     'share/doc/YaTeA/Makefile.in' => 'share/doc/YaTeA/Makefile.in',
		     'share/doc/YaTeA/pat_decomp.fig' => 'share/doc/YaTeA/pat_decomp.fig',
		     'share/doc/YaTeA/PlaceInsert.fig' => 'share/doc/YaTeA/PlaceInsert.fig',
		     'share/doc/YaTeA/samples/pmid10788508-v2.genia2.xml' => 'share/doc/YaTeA/samples/pmid10788508-v2.genia2.xml',
		     'share/doc/YaTeA/SatureEtMAJ.fig' => 'share/doc/YaTeA/SatureEtMAJ.fig',
		     'share/doc/YaTeA/TAtt.fig' => 'share/doc/YaTeA/TAtt.fig',
		     'share/doc/YaTeA/TechnicalDocYatea.bbl' => 'share/doc/YaTeA/TechnicalDocYatea.bbl',
		     'share/doc/YaTeA/TechnicalDocYatea.tex' => 'share/doc/YaTeA/TechnicalDocYatea.tex',
		     'share/doc/YaTeA/transfig.tex' => 'share/doc/YaTeA/transfig.tex',
		     'share/doc/YaTeA/UserManualForTechDoc.tex' => 'share/doc/YaTeA/UserManualForTechDoc.tex',
		     'share/doc/YaTeA/Yatea_Classes_Config.fig' => 'share/doc/YaTeA/Yatea_Classes_Config.fig',
		     'share/doc/YaTeA/Yatea_Classes_Process.fig' => 'share/doc/YaTeA/Yatea_Classes_Process.fig',
		     'share/doc/YaTeA/Yatea_user_manual.tex' => 'share/doc/YaTeA/Yatea_user_manual.tex',
		     'share/YaTeA/config/EN/ChunkingExceptions' => 'share/YaTeA/config/EN/ChunkingExceptions',
		     'share/YaTeA/config/EN/ChunkingFrontiers' => 'share/YaTeA/config/EN/ChunkingFrontiers',
		     'share/YaTeA/config/EN/CleaningExceptions' => 'share/YaTeA/config/EN/CleaningExceptions',
		     'share/YaTeA/config/EN/CleaningFrontiers' => 'share/YaTeA/config/EN/CleaningFrontiers',
		     'share/YaTeA/config/EN/ForbiddenStructures' => 'share/YaTeA/config/EN/ForbiddenStructures',
		     'share/YaTeA/config/EN/LGPmapping' => 'share/YaTeA/config/EN/LGPmapping',
		     'share/YaTeA/config/EN/LinguisticConfiguration' => 'share/YaTeA/config/EN/LinguisticConfiguration',
		     'share/YaTeA/config/EN/Options' => 'share/YaTeA/config/EN/Options',
		     'share/YaTeA/config/EN/ParsingPatterns' => 'share/YaTeA/config/EN/ParsingPatterns',
		     'share/YaTeA/config/EN/TagSet' => 'share/YaTeA/config/EN/TagSet',
		     'share/YaTeA/config/FR/ChunkingExceptions' => 'share/YaTeA/config/FR/ChunkingExceptions',
		     'share/YaTeA/config/FR/ChunkingFrontiers' => 'share/YaTeA/config/FR/ChunkingFrontiers',
		     'share/YaTeA/config/FR/CleaningExceptions' => 'share/YaTeA/config/FR/CleaningExceptions',
		     'share/YaTeA/config/FR/CleaningFrontiers' => 'share/YaTeA/config/FR/CleaningFrontiers',
		     'share/YaTeA/config/FR/CoordPatterns' => 'share/YaTeA/config/FR/CoordPatterns',
		     'share/YaTeA/config/FR/ForbiddenStructures' => 'share/YaTeA/config/FR/ForbiddenStructures',
		     'share/YaTeA/config/FR/LinguisticConfiguration' => 'share/YaTeA/config/FR/LinguisticConfiguration',
		     'share/YaTeA/config/FR/mapping.TT2Lexter' => 'share/YaTeA/config/FR/mapping.TT2Lexter',
		     'share/YaTeA/config/FR/Options' => 'share/YaTeA/config/FR/Options',
		     'share/YaTeA/config/FR/ParsingPatterns' => 'share/YaTeA/config/FR/ParsingPatterns',
		     'share/YaTeA/config/FR/TagSet' => 'share/YaTeA/config/FR/TagSet',
		     'share/YaTeA/config/FR-Flemm/ChunkingExceptions' => 'share/YaTeA/config/FR-Flemm/ChunkingExceptions',
		     'share/YaTeA/config/FR-Flemm/ParsingPatterns' => 'share/YaTeA/config/FR-Flemm/ParsingPatterns',
		     'share/YaTeA/config/FR-Flemm/ForbiddenStructures' => 'share/YaTeA/config/FR-Flemm/ForbiddenStructures',
		     'share/YaTeA/config/FR-Flemm/ChunkingFrontiers' => 'share/YaTeA/config/FR-Flemm/ChunkingFrontiers',
		     'share/YaTeA/config/FR-Flemm/Options' => 'share/YaTeA/config/FR-Flemm/Options',
		     'share/YaTeA/config/FR-Flemm/CleaningFrontiers' => 'share/YaTeA/config/FR-Flemm/CleaningFrontiers',
		     'share/YaTeA/config/FR-Flemm/CleaningExceptions' => 'share/YaTeA/config/FR-Flemm/CleaningExceptions',
		     'share/YaTeA/config/FR-Flemm/TagSet' => 'share/YaTeA/config/FR-Flemm/TagSet',
		     'share/YaTeA/locale/EN/Messages' => 'share/YaTeA/locale/EN/Messages',
		     'share/YaTeA/locale/FR/Messages' => 'share/YaTeA/locale/FR/Messages',
		     'share/YaTeA/samples/bootstrap.testified' => 'share/YaTeA/samples/bootstrap.testified',
		     'share/YaTeA/samples/corpus.ttg' => 'share/YaTeA/samples/corpus.ttg',
		     'share/YaTeA/samples/environment_EN.TT' => 'share/YaTeA/samples/environment_EN.TT',
		     'share/YaTeA/samples/pmid10788508-v2.1l' => 'share/YaTeA/samples/pmid10788508-v2.1l',
		     'share/YaTeA/samples/pmid10788508-v2.TT' => 'share/YaTeA/samples/pmid10788508-v2.TT',
		     'share/YaTeA/samples/sampleEN.ttg' => 'share/YaTeA/samples/sampleEN.ttg',
		     'share/YaTeA/samples/termes_environnement.ttg' => 'share/YaTeA/samples/termes_environnement.ttg',
		     'share/YaTeA/samples/termes_environnement.txt' => 'share/YaTeA/samples/termes_environnement.txt',
		     'share/YaTeA/samples/termlist' => 'share/YaTeA/samples/termlist',
		     'share/YaTeA/samples/termlist.TT' => 'share/YaTeA/samples/termlist.TT',
		     'share/YaTeA/samples/termsEnvironment.ttg' => 'share/YaTeA/samples/termsEnvironment.ttg',
		     'share/YaTeA/samples/termsEnvironment.txt' => 'share/YaTeA/samples/termsEnvironment.txt',
		     'share/YaTeA/samples/termsPhB.termsgt' => 'share/YaTeA/samples/termsPhB.termsgt',
		     'share/YaTeA/samples/transcript-epure-Genia.bootstrap' => 'share/YaTeA/samples/transcript-epure-Genia.bootstrap',
		     'share/YaTeA/yatea.conf' => 'share/YaTeA/yatea.conf',
		 },
     );

$build->install_base_relpaths( conf => 'etc' );
$build->prefix_relpaths('site', conf => 'etc' );
$build->prefix_relpaths('core', conf => 'etc' );
$build->prefix_relpaths('vendor', conf => 'etc' );
$build->add_build_element('conf');


$build->install_base_relpaths( share => 'share' );
$build->prefix_relpaths('site', share => 'share' );
$build->prefix_relpaths('core', share => 'share' );
$build->prefix_relpaths('vendor', share => 'share' );
$build->add_build_element('share');

foreach my $type (keys(%{$build->install_sets})) {
    $build->install_sets->{$type}->{'conf'} = '/etc';
    $build->install_sets->{$type}->{'share'} = '/usr/share';
}

$build->install_sets->{'site'}->{'share'} = '/usr/local/share';

$build->install_sets->{'site'}->{'conf'} = '/usr/local/etc';

$build->install_path->{'conf'} = '/etc';
$build->install_path->{'share'} = '/usr/share';

# same thing with prefix
if (((defined $build->install_base()) && ($build->install_base() eq "/usr")) ||
    ((defined $build->prefix()) && ($build->prefix() eq "/usr")))
 {
	$build->install_path->{'conf'} = '/etc';
	$build->install_path->{'share'} = '/usr/share';
}


$build->create_build_script;


